File: 73-set-matrix-zeroes/set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */


var setZeroes = function (matrix) {
    let row = matrix.length;
    let col = matrix[0].length;

    if (row === 0 || col === 0) {
        return matrix;
    }

    function markRow(i) {
        for (let j = 0; j < col; j++) {
            if (matrix[i][j] != 0) {
                matrix[i][j] = Number.MIN_SAFE_INTEGER;
            }
        }
    }

    function markCol(j) {
        for (let i = 0; i < row; i++) {
            if (matrix[i][j] != 0) {
                matrix[i][j] = Number.MIN_SAFE_INTEGER;
            }
        }
    }

    for (let i = 0; i < row; i++) {
        for (let j = 0; j < col; j++) {
            if (matrix[i][j] === 0) {
                markRow(i);
                markCol(j);
            }
        }
    }

    for (let i = 0; i < row; i++) {
        for (let j = 0; j < col; j++) {
            if (matrix[i][j] === Number.MIN_SAFE_INTEGER) {
                matrix[i][j] = 0;
            }
        }
    }

    return matrix;
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int rowSize = matrix.length;
        int colSize = matrix[0].length;

        int[] row = new int[rowSize]; 
        int[] col = new int[colSize]; 

        for(int i =0; i<rowSize; i++){
            for(int j =0; j<colSize; j++){
                if(matrix[i][j] == 0){
                    row[i] = Integer.MIN_VALUE;
                    col[j] = Integer.MIN_VALUE;
                }
                    }
        }

        for(int i =0; i<rowSize; i++){
            for(int j =0; j<colSize; j++){
                if(row[i] ==Integer.MIN_VALUE || col[j] == Integer.MIN_VALUE ){
                    matrix[i][j] = 0;
                }
                    }
        }

    }
}